csharp.test/TestLogicalTypeRoundtrip.cs

```diff
@@ -288,6 +288,49 @@ public static void TestNestedStructArray([Values(Repetition.Required, Repetition
             fileReader.Close();
         }
 
+        [Test]
+        public static void TestLargeArraysEnumerator()
+        {
+            CheckEnumerator(Enumerable.Range(0, 4100).ToArray());
+            CheckEnumerator(Enumerable.Range(0, 4100).Select(i => new[] {$"row {i}"}).ToArray());
```

Comment:
This test works on the assumption that the default BufferLength=4096. 
Might be worth explicitly setting that.

Reply:
Good idea, specified the buffer length to use and added a comment.